lib/context.js

```diff
@@ -1041,19 +1041,107 @@ function _expandIri(activeCtx, value, relativeTo, localCtx, defined, options) {
     }
   }
 
-  // prepend vocab
   if(relativeTo.vocab && '@vocab' in activeCtx) {
-    return activeCtx['@vocab'] + value;
-  }
+    // prepend vocab
+    const prependedResult = activeCtx['@vocab'] + value;
+    let expansionMapResult = undefined;
+    if(options && options.expansionMap) {
+      // if we are about to expand the value by prepending
+      // @vocab then call the expansion map to inform
+      // interested callers that this is occurring
+
+      // TODO: use `await` to support async
+      expansionMapResult = options.expansionMap({
+        prependedIri: {
+          type: '@vocab',
+          vocab: activeCtx['@vocab'],
+          value,
+          result: prependedResult
+        },
+        activeCtx,
+        options
+      });
 
-  // prepend base
-  if(relativeTo.base && '@base' in activeCtx) {
+    }
+    if(expansionMapResult !== undefined) {
+      value = expansionMapResult;
+    } else {
+      // the null case preserves value as potentially relative
+      value = prependedResult;
+    }
+  } else if(relativeTo.base && '@base' in activeCtx) {
+    // prepend base
     if(activeCtx['@base']) {
-      // The null case preserves value as potentially relative
-      return prependBase(prependBase(options.base, activeCtx['@base']), value);
+      const prependedResult = prependBase(
+        prependBase(options.base, activeCtx['@base']), value);
+
+      let expansionMapResult = undefined;
+      if(options && options.expansionMap) {
+        // if we are about to expand the value by prepending
+        // @base then call the expansion map to inform
+        // interested callers that this is occurring
+
+        // TODO: use `await` to support async
+        expansionMapResult = options.expansionMap({
+          prependedIri: {
+            type: '@base',
+            base: activeCtx['@base'],
+            value,
+            result: prependedResult
+          },
+          activeCtx,
+          options
+        });
+      }
+      if(expansionMapResult !== undefined) {
+        value = expansionMapResult;
+      } else {
+        // the null case preserves value as potentially relative
+        value = prependedResult;
+      }
     }
   } else if(relativeTo.base) {
-    return prependBase(options.base, value);
+    const prependedResult = prependBase(options.base, value);
+    let expansionMapResult = undefined;
+    if(options && options.expansionMap) {
+      // if we are about to expand the value by prepending
+      // @base then call the expansion map to inform
+      // interested callers that this is occurring
+
+      // TODO: use `await` to support async
+      expansionMapResult = options.expansionMap({
+        prependedIri: {
```

Comment:
Is there any doc on expansionMap that what data will be passed to the hook? Do users have to read the code or unit tests to understand?

Reply:
Nothing at the moment, but PRs welcome to add something to the README.